### PR TITLE
ARROW-14189: [Docs] Add version dropdown to the sphinx docs

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
There is working happening for this in the upstream theme (https://github.com/pydata/pydata-sphinx-theme/pull/436), but since that is not yet merged/released, I will "backport" it to our own docs.

The basics already work here:

Peek 2021-10-01 11-46

Some notes:

- This is the basic, default styling of a dropdown from bootstrap (using the bootstrap primary color). We might want to style it a bit differently to have it fit better with our color scheme 
- Exact placement of the dropdown on the page can be improved
- You can go back to earlier versions (and it preserves the page you are on), but those earlier versions don't have this dropdown (unless we would rebuild the older docs with this patch).

File: docs/source/_templates/version-switcher.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<!-- Copied from https://github.com/pydata/pydata-sphinx-theme/pull/436
+TODO update this once that is included in the upstream theme -->
+
+<div class="dropdown">
+    <button type="button" class="btn btn-primary btn-sm navbar-btn dropdown-toggle" id="switcher_button" data-toggle="dropdown">
+        {{ switcher_prefix }}{{ release }}
```
Comment (pitrou) [inline docs/source/_templates/version-switcher.html:6]: I'm curious: where does `switcher_prefix` come from?

Comment (jorisvandenbossche) [reply docs/source/_templates/version-switcher.html:6]: Ah, I forgot to check if I could remove it (I asked exactly the same on the upstream PR :))

**Comment by kszucs** [comment]
Thanks @jorisvandenbossche for working on this!

We need to update the release script to create and [mount the right directory](https://github.com/apache/arrow/blob/master/dev/release/post-09-docs.sh#L47) to the docker container which builds the documentation:

```bash
mkdir -p ${ARROW_SITE_DIR}/docs/${version}
# ...
-v "${ARROW_SITE_DIR}/docs/${version}:/build/docs"
```

We either need to use full version numbers in the selector or trim the version number in the release script to major and minor parts.

Other than that can we merge this PR?

**Review by kszucs — APPROVED** [review]
Thanks Joris!

**Pull request closed without merging**